### ISSUE TITLE
MMHQ -> MMLab -> java.lang.NoClassDefFoundError: org/w3c/dom/svg/SVGDocument

### ISSUE THREAD

**Issue by Membooo** [MegaMek/megameklab#1263]
Fill out the fields below with as much pertinent information as possible to help the developers solve the issue.
Search for your issue inside open and closed issues first before reporting so as not to make duplicates.
(Delete the above lines)

### Environment
MMHQs Lab:
Windows 11
OpenJDK 11
	Today: 2023-05-25
	Origin Project: MegaMekLab
	Java Vendor: Eclipse Adoptium
	Java Version: 11.0.15
	Platform: Windows 11 10.0 (amd64)
	System Locale: de_DE

### Description
When Printing error occurs

### Files
$JAXPSAXParser.parse(SAXParserImpl.java:635)
	at org.apache.batik.dom.util.SAXDocumentFactory.createDocument(SAXDocumentFactory.java:453)
	at org.apache.batik.dom.util.SAXDocumentFactory.createDocument(SAXDocumentFactory.java:287)
	at megameklab.printing.PrintRecordSheet.loadSVG(PrintRecordSheet.java:241)
	at megameklab.printing.PrintRecordSheet.loadTemplate(PrintRecordSheet.java:280)
	at megameklab.printing.PrintRecordSheet.createDocument(PrintRecordSheet.java:287)
	at megameklab.printing.PrintRecordSheet.print(PrintRecordSheet.java:323)
	at java.desktop/sun.print.RasterPrinterJob.printPage(RasterPrinterJob.java:2275)
	at java.desktop/sun.print.RasterPrinterJob.print(RasterPrinterJob.java:1654)
	at megameklab.printing.RecordSheetTask$PrintTask.doInBackground(RecordSheetTask.java:160)
	at megameklab.printing.RecordSheetTask$PrintTask.doInBackground(RecordSheetTask.java:137)
	at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.ClassNotFoundException: org.w3c.dom.svg.SVGDocument
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 41 more

![printting_issue](https://github.com/MegaMek/megameklab/assets/113114257/7299c8a1-47bd-4da7-9141-8bcfde3d8058)

**Comment by HammerGS** [comment]
@HoneySkull Getting this error reported in 0.49.13 could this be related to the Gradle PR you submitted.

**Comment by HoneySkull** [comment]
I'm taking a look.  This occurs when trying to print?

**Comment by HammerGS** [comment]
It does and I can duplicate this on 49.13 windows release

**Comment by HoneySkull** [comment]
Confirmed while running with Java 11 runtime.

**Comment by HoneySkull** [comment]
I created a new megamek build after backing out the gradle file change and the print functions with that build it is definitely that change that broke it.  @HammerGS - I can back these changes out of megemeklab and mekhq.  I'm wondering if there is a way to move these 'exclude' statement to `settings_local.gradle' or local properties so Eclipse can still be used but are not excluded as part of the build/release process.

**Comment by HammerGS** [comment]
I know there has been a long standing issue around these on the Eclipse issues tracker. I’m open to any options.

**Comment by HoneySkull** [comment]
[memamek_print_workaround.zip](https://github.com/MegaMek/megameklab/files/11570730/memamek_print_workaround.zip)

For anyone who really needs to print mechs before a fix:  You can replace the memamek.exe launch4j (needed for class path) and the xml-apis jar in the `lib` folder.  

xml-apis-1.4.01.jar -> lib
xml-apis-ext-1.3.04.jar ->lib
megameklab.exe -> main folder.

**Comment by HoneySkull** [comment]
[mekhq_print_workaround.zip](https://github.com/MegaMek/megameklab/files/11570898/mekhq_print_workaround.zip)

Added the mekhq.exe to enable printing units in MekHQ from the meklab or from the briefing tab.

**Comment by HammerGS** [comment]
Thanks. I’ll test on my system in the morning and update the release posts with these.

**Comment by me0262** [comment]
Confirmed this also affects Linux+OpenJDK 17.0.6 and MacOS 10.15+Oracle JRE 17.0.5

Confirmed that HoneySkull's fix works on Linux+OpenJDK 17.0.6.
Oddly, neither .jar files are present from a fresh download, but in the mml-startup script, xml-apis-ext.jar is present in CLASSPATH on line 67, but not xml-apis.jar, which has to be added at the end as `:$APP_HOME/lib/xml-apis-ext-1.3.04.jar`

**Comment by HoneySkull** [comment]
The Eclipse IDE Runtime environment has a conflict because both the Java Runtime xml-apis dependencies use the same org.w3c.dom name space.  The fix to the gradle build file that I put into 0.49.13 that broke printing excluded the xml-apis group from the release assembly process.  That's what caused the two xml-apis jar files to be missing from the distribution.

I believe I found a fix that will exclude ONLY the xml-apis but still include the xml-apis-ext jar in the build process.  However, I need to go through each instance of multiple dependency in the code to verify nothing is broken because of it.

The printing was broken because the xml-apis-ext jar is used for all the SVG rendering when printing and cannot be excluded.  While the org.w3c.dom is duplicated in Java Runtime, the namespace in the xml-apis-ext is not.

The change I made inadvertently excluded the xml-apis **group** not just the core xml-apis that should to be excluded.

In short;  if xml-apis is excluded but xml-apis-ext IS still included - everything tested so far seems to work both in Eclipse IDE as well as the final build.   But I have a lot more testing to do to make sure some other edge cases are not broken.

If anyone wants to help me test a candidate fix, let me know.  I would love to continue to use Eclipse as an IDE.  BTW:  I've very sorry about the timing of breaking printing in the .13 build.  I didn't know it was about to be released.

**Comment by HammerGS** [comment]
No worries on the breaking stuff. It’s what devs are for. 

For testing I prefer Eclipse and build the releases. If you want to open a draft PR I can pull and test. Which I think would be ideal as I’m the release guy and on Windows

**Comment by HoneySkull** [comment]
Draft PR submitted.  If this checkout, I can apply a draft PR to MekQH as well.

**Comment by HoneySkull** [comment]
Fixed by PR #1270

**Comment by SJuliez** [comment]
Closing this as it is (hopefully) fixed
